.github/workflows/install-sfdx-package.yml

```
@@ -244,3 +244,21 @@ jobs:
 
       - name: install package
         run: sf package install --package "${package}" --installation-key "${installation_key_decrypted}" --apex-compile package --upgrade-type Mixed --wait 15
+
+      - name: check out code
+        uses: actions/checkout@v4
+
+      - name: find post-deploy scripts
+        id: find-post-deploy-scripts
+        run: |
+          mkdir -p deploy/hooks/post-deploy
+          ls deploy/hooks/post-deploy/*.sh > post_deploy_scripts.txt || echo "No post-deploy scripts found" > post_deploy_scripts.txt
+
+      - name: execute post-deploy scripts
+        if: ${{ env.post_deploy_scripts != '' }}
+        run: |
+          while IFS= read -r script; do
+            if [ -f "$script" ]; then
+              bash "$script"
```

OK so this one's tricky. 

- The sub-shell will inherit stdin from the parent shell. 
- Stdin in our parent shell is reading from `post_deploy_scripts.txt`. 
- So if our post-deploy script tries to read from stdin it will get the remaining data from `post_deploy_scripts.txt` instead ... and our while loop would finish prematurely because there's nothing left to read; so not all the post-deploy scripts would actually be run then.

So we want our sub-shell to read from something else so it doesn't eat up our stdin. There's nothing to read from, so `/dev/null` should do.

```suggestion
              bash "$script"</dev/null
```